src/main/java/com/iexec/worker/result/ResultService.java

```diff
@@ -1,104 +1,97 @@
                          PublicConfigurationService publicConfigService,
                          CredentialsService credentialsService,
                          IexecHubService iexecHubService,
-                         CustomResultFeignClient customResultFeignClient) {
+                         CustomResultFeignClient customResultFeignClient,
+                         ComputationService computationService) {
         this.workerConfigService = workerConfigService;
         this.publicConfigService = publicConfigService;
         this.credentialsService = credentialsService;
         this.iexecHubService = iexecHubService;
         this.customResultFeignClient = customResultFeignClient;
+        this.computationService = computationService;
         this.resultInfoMap = new ConcurrentHashMap<>();
     }
 
-    public boolean saveResult(String chainTaskId, TaskDescription taskDescription, String stdout) {
-        try {
-            saveStdoutFileInResultFolder(chainTaskId, stdout);
-            zipResultFolder(chainTaskId);
-            saveResultInfo(chainTaskId, taskDescription);
-            return true;
-        } catch (Exception e) {
-            return false;
-        }
+    public ResultInfo getResultInfos(String chainTaskId) {
+        return resultInfoMap.get(chainTaskId);
     }
 
-    private File saveStdoutFileInResultFolder(String chainTaskId, String stdoutContent) {
-        log.info("Stdout file added to result folder [chainTaskId:{}]", chainTaskId);
-        String filePath = getResultFolderPath(chainTaskId) + File.separator + STDOUT_FILENAME;
-        return createFileWithContent(filePath, stdoutContent);
+    public String getResultFolderPath(String chainTaskId) {
+        return workerConfigService.getTaskIexecOutDir(chainTaskId);
     }
 
-    public void zipResultFolder(String chainTaskId) {
-        File zipFile = FileHelper.zipFolder(getResultFolderPath(chainTaskId));
-        log.info("Zip file has been created [chainTaskId:{}, zipFile:{}]", chainTaskId, zipFile.getAbsolutePath());
+    public boolean isResultFolderFound(String chainTaskId) {
+        return new File(getResultFolderPath(chainTaskId)).exists();
     }
 
-    public void saveResultInfo(String chainTaskId, TaskDescription taskDescription) {
-        ComputedFile computedFile = getComputedFile(chainTaskId);
+    public String getResultZipFilePath(String chainTaskId) {
+        return getResultFolderPath(chainTaskId) + ".zip";
+    }
+
+    public String getEncryptedResultFilePath(String chainTaskId) {
+        return getResultFolderPath(chainTaskId) + ".zip";
+    }
+
+    public boolean isResultZipFound(String chainTaskId) {
+        return new File(getResultZipFilePath(chainTaskId)).exists();
+    }
 
+    public boolean isEncryptedResultZipFound(String chainTaskId) {
+        return new File(getEncryptedResultFilePath(chainTaskId)).exists();
+    }
+
+    public void saveResultInfo(String chainTaskId, TaskDescription taskDescription) {
```

Comment:
Note for the future: We might see if we can get rid of ResultInfo. I'm not sure this is still relevant

Reply:
I agree yes, but in another iteration maybe.